src/mtev_conf.c

```diff
@@ -1100,21 +1100,50 @@ static int mtev_conf_check_value(mtev_conf_description_t* description) {
   return rv;
 }
 
+char*
+mtev_conf_section_to_xpath(mtev_conf_section_t* section) {
+  if (section == NULL) {
+    return NULL;
+  }
+
+  const int max_len = 512;
+  char* string = alloca(max_len);
```

Comment:
I think 512 is a tad shy... you should run two loops through, the first time to determine the size.